### PR TITLE
Migrate LanguageUtil to kotlin

### PR DESCRIPTION
**Pull request by lukstbit**
## Purpose / Description
This PR migrates the LanguageUtil.java file to kotlin following the kotlin migration guide. 

Info below not applicable, will go in another PR:
_Also:
We currently have two classes for language related utilities: [LanguageUtil](https://github.com/ankidroid/Anki-Android/blob/7082f6ee93ed165180b935dbca44598a6566b3cc/AnkiDroid/src/main/java/com/ichi2/utils/LanguageUtil.java) and [LanguageUtils](https://github.com/ankidroid/Anki-Android/blob/0d663ae432ca98f46236012c310625462432e541/AnkiDroid/src/main/java/com/ichi2/anki/LanguageUtils.kt). This doesn't make sense so in the last commit I combined the two classes into one in the utils package. If this is not ok please ignore the last commit._

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by ShridharGoel — CHANGES_REQUESTED** [review]
The last commit would be better in a different PR.

**Comment by lukstbit** [comment]
@ShridharGoel This PR has only the two migration commits now, I'll create another PR for the changes in the last commit.

File: AnkiDroid/src/main/java/com/ichi2/utils/LanguageUtil.kt [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import java.text.DateFormat
+import java.util.*
+
+/**
+ * Utility call for proving language related functionality.
+ */
+object LanguageUtil {
+    /** A list of all languages supported by AnkiDroid
+     * Please modify LanguageUtilsTest if changing
+     * Please note 'yue' is special, it is 'yu' on crowdin, and mapped in import specially to 'yue'  */
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/utils/LanguageUtil.kt:34]:
```suggestion
     * Please note 'yue' is special, it is 'yu' on Crowdin, and mapped in import specially to 'yue'  */
```

The conflict it took is just the typo change from here https://github.com/ankidroid/Anki-Android/commit/5da9561a290122b52047c5c24648f078dc02dca7#diff-4ab81775efd50ce812d06d139972eb25292efd73500eb86efdb15027ad453779 - easy to ingest in the migration and mark as de-conflicted

**Comment by mikehardy** [comment]
Hmm - I am not sure how exactly to fix this one, it appears that the first commit needs a rebase against main to clear the conflict from a git perspective

**Comment by lukstbit** [comment]
@mikehardy The quick fix didn't work, I manually made the change. It is even better this way because the PR doesn't get an unrelated commit.

File: AnkiDroid/src/main/java/com/ichi2/utils/LanguageUtil.kt [hunk 2]
```diff
@@ -13,109 +13,99 @@
...
 
 /**
  * Utility call for proving language related functionality.
  */
-public class LanguageUtil {
-
+object LanguageUtil {
     /** A list of all languages supported by AnkiDroid
      * Please modify LanguageUtilsTest if changing
      * Please note 'yue' is special, it is 'yu' on CrowdIn, and mapped in import specially to 'yue' */
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/utils/LanguageUtil.kt:34]: Verified it contains the change that blew this up with the conflict:
https://github.com/ankidroid/Anki-Android/pull/10723/files#diff-4ab81775efd50ce812d06d139972eb25292efd73500eb86efdb15027ad453779R42

:100:

**Review by mikehardy — APPROVED** [review]
Looks great to me now, and I think Shrdidhars comment was addressed
Ready for someone else to look at it

**Review by mikehardy — APPROVED** [review]
looks good to go, thanks!

**Pull request merged**